W-14069450-botsessioninformation-se

sebermuley opened the pull request:
**Replacement of Broken-W-14069450-botsessioninformation-se**
**Already SME reviewed

Added and restructured information on Secure Sessions, so that user's don't need to wonder anymore and support cases are hopefully reduced.
New Topic: RPA Home > Security > Secure Sessions
Changed Topic: RPA Bot > System and Hardware Requirements

Commit:
...

Commit:
New topic Security - Secure Sessions in RPA Home

Commit:
...

Commit:
...

Commit:
Secure Sessions and System Requirements
ready for SME review

Commit:
implemented suggestions from SME review

Commit:
Merge branch 'latest' into W-14069450-botsessioninformation-se

Commit:
started reworking of RPA Bot Configurator

Commit:
Configuring RPA Bot
inkl Process Streaming
ready for SME review

Commit:
...

Commit:
removed 'Recommended' for System Settings

Commit:
implemented suggestions from SME review

Commit:
moved changes from bot 1.3.4
to 1.3

Commit:
Merge branch 'latest' into bypassmergelatest

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -1,5 +1,7 @@
 == Hardware Requirements
 
+* We recommend using a dedicated computer for running the RPA Bot.
```
Our [style](https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22MuleSoftrecommends%22(%22advises,%22%22suggests%22%E2%80%A6)) is to avoid "we recommend"

Suggest something like "For optimal performance (or whatever the reason behind the recommendation is), use a dedicated....

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+* Display resolution: 
+** Minimum: 1280 x 1080
+** Recommended: Full HD 1080p (1920 × 1080) or higher
+* Scaling of text and other items on the screen: 100 %
```
```suggestion
* Scaling of text and other items on the screen: 100%
```

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+* Display resolution: 
+** Minimum: 1280 x 1080
+** Recommended: Full HD 1080p (1920 × 1080) or higher
+* Scaling of text and other items on the screen: 100 %
```
Is there a name for this in system settings? Object scaling percentage?

sebermuley replied:
It is called 'custom scaling' in the OS. I found 'font choices' on our style page and chose Monospaced font for the Windows system setting names now. I hope this is ok... (See Slack for Screenshots.)

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+** Minimum: 1280 x 1080
+** Recommended: Full HD 1080p (1920 × 1080) or higher
+* Scaling of text and other items on the screen: 100 %
+* _Clear Type_: turned off
```
Why italics? Is this a system setting?

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+** Minimum: 1280 x 1080
+** Recommended: Full HD 1080p (1920 × 1080) or higher
+* Scaling of text and other items on the screen: 100 %
+* _Clear Type_: turned off
```
```suggestion
* _Clear Type_: disabled
```

sebermuley replied:
The Windows UI uses 'turn on' -> I'm leaving that (BTW I don't think it's confusing in general for the reader to read phrasal verbs)

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+
+=== Requirements for the Secure Session Technology
+
+Secure Sessions are hidden Desktop sessions. They run in the background and only the bot can access them. An additional advantage of using Secure Sessions is, that the bot can multiply itsself to run several process instances in parallel.
```
```suggestion
Secure Sessions are hidden Desktop sessions that run in the background and are accessible only to the bot. An additional advantage of using Secure Sessions is that the bot can multiply itself to run several process instances in parallel.
```

hannanelson replied:
I'm breaking the cardinal rue of active voice here :(

hannanelson replied:
You mention an additional advantage, but I don't see the original advantage.

This paragraph is really about what Secure Sessions are and the advantages, rather than requirements.

sebermuley replied:
No problem for me. I like passive voice :)

sebermuley replied:
Yes, I included the explanation of what Secure Sessions are because the problem seems to be that our users can't seem to find that information easily.

1st advantage: hidden
2nd advantage: parallel

hannanelson replied:
Ah, I missed that hidden=advantage :>

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+
+Secure Sessions are hidden Desktop sessions. They run in the background and only the bot can access them. An additional advantage of using Secure Sessions is, that the bot can multiply itsself to run several process instances in parallel.
+
+To use Secure Sessions you must ensure that all prerequisites are met and all responsible persons are informed.
```
Any way to make this active voice?

hannanelson replied:
I wonder if you can make this lowercase in this case? It's kind of like:
connected apps, the Connected Apps feature
Lowercase the common noun and uppercase the proper noun.

sebermuley replied:
The problem is that it is not defined who does this and in what way.

sebermuley replied:
I think we only use it as proper noun.
There is a patent and I think the name is registered.

hannanelson replied:
Suggest reviewing all mentions and making sure they're correct.

sebermuley replied:
I upper-cased all occurrences of Secure Session and Desktop Session, because this is Servicetrace terminology.
see https://techcommunity.microsoft.com/t5/ask-the-performance-team/sessions-desktops-and-windows-stations/ba-p/372473 for general information about 'Sessions'
Desktop Session = "consists of all of the processes and other system objects that represent a single user’s logon session." and the user is currently logged in. 
Secure Session = same but hidden. Because it is a Remote Desktop Session at the local loopback address.
@Cristian-Venticinque maybe we can add a user-friendly definition of these two terms to the Glossary

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+
+Secure Sessions are hidden Desktop sessions. They run in the background and only the bot can access them. An additional advantage of using Secure Sessions is, that the bot can multiply itsself to run several process instances in parallel.
+
+To use Secure Sessions you must ensure that all prerequisites are met and all responsible persons are informed.
```
```suggestion
Before you implement Secure Sessions, first meet all prerequisites and inform all responsible persons:
```

sebermuley replied:
The person reading this can not meet all the prerequisites and might not know who the responsible persons are. I think this is part of the support problem... 
With that text we can at least raise awareness I hope.

hannanelson replied:
What about:
Before you implement Secure Sessions, first ensure all prerequisites are met and inform all responsible persons:

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
-* 64-bit Operating System.
-* Dedicated Windows Server machine with enough resources to execute several processes simultaneously (2 GB or more RAM per session)
-* The RPA Bot service runs under the _Local System account_ (by default, the setup configures RPA Bot to use this account).
+* The RPA Bot service runs under the _Local System_ account per default.
```
Why italics?

sebermuley replied:
It is a name used by Windows, see other comment

sebermuley replied:
replaced with monospaced, see Slack

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
-* Dedicated Windows Server machine with enough resources to execute several processes simultaneously (2 GB or more RAM per session)
-* The RPA Bot service runs under the _Local System account_ (by default, the setup configures RPA Bot to use this account).
+* The RPA Bot service runs under the _Local System_ account per default.
+** If you want to use another account, ensure that it has permission to run Windows services.
```
```suggestion
** To use another account, ensure that it has permission to run Windows services.
```

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
-* The RPA Bot service runs under the _Local System account_ (by default, the setup configures RPA Bot to use this account).
+* The RPA Bot service runs under the _Local System_ account per default.
+** If you want to use another account, ensure that it has permission to run Windows services.
+* The RPA Bot service runs processes under user accounts existing on the same computer or within the same domain.
```
```suggestion
* The RPA Bot service runs processes under user accounts on the same computer or within the same domain.
```

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+* The RPA Bot service runs under the _Local System_ account per default.
+** If you want to use another account, ensure that it has permission to run Windows services.
+* The RPA Bot service runs processes under user accounts existing on the same computer or within the same domain.
+** We recommend using dedicated accounts and a user naming scheme to differentiate between local users and domain users and between human users and bot users.
```
```suggestion
** Use dedicated accounts and a user-naming scheme to differentiate between local users and domain users and between human users and bot users.
```

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+** If you want to use another account, ensure that it has permission to run Windows services.
+* The RPA Bot service runs processes under user accounts existing on the same computer or within the same domain.
+** We recommend using dedicated accounts and a user naming scheme to differentiate between local users and domain users and between human users and bot users.
+** User accounts used by the bot to run processes must belong to the _Remote Desktop Users_ group and must be allowed to establish a local RDP session via the local loopback address 127.0.0.2.
```
Our style is to use italics for emphasis (sparingly) and variable names.

sebermuley replied:
replaced with monospaced font, see Slack

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+** If you want to use another account, ensure that it has permission to run Windows services.
+* The RPA Bot service runs processes under user accounts existing on the same computer or within the same domain.
+** We recommend using dedicated accounts and a user naming scheme to differentiate between local users and domain users and between human users and bot users.
+** User accounts used by the bot to run processes must belong to the _Remote Desktop Users_ group and must be allowed to establish a local RDP session via the local loopback address 127.0.0.2.
```
```suggestion
** User accounts used by the bot to run processes must belong to the _Remote Desktop Users_ group and must have permission to establish a local RDP session via the local loopback address 127.0.0.2.
```

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+* The RPA Bot service runs processes under user accounts existing on the same computer or within the same domain.
+** We recommend using dedicated accounts and a user naming scheme to differentiate between local users and domain users and between human users and bot users.
+** User accounts used by the bot to run processes must belong to the _Remote Desktop Users_ group and must be allowed to establish a local RDP session via the local loopback address 127.0.0.2.
+* If you want a bot to run more than one (Windows 10, Windows 11) or two (Windows Server) Secure Sessions simultaneously, make sure that a terminal server software with enough licensed users is installed on the computer where the bot is installed. The computer must have enough resources to execute several processes simultaneously: 2 GB or more RAM per session. 
```
```suggestion
* To enable a bot to run more than one (Windows 10, Windows 11) or two (Windows Server) secure sessions simultaneously, ensure that terminal server software with enough licensed users is installed on the computer where the bot is installed. The computer must have enough resources to execute several processes simultaneously: 2 GB or more RAM per session. 
```

hannanelson replied:
What does enough (licensed users) mean in this case? How do they determine that?

hannanelson replied:
The computer must have at least 2 GB of RAM to run each secure session?

Are processes = sessions? Or secure sessions = several processes?

sebermuley replied:
The Terminal Server (simulation) software enables the operting system to run more than one (or two) sessions in parallel.
A session is what you get when you log in to the computer.  You have a Desktop that you can see and the system to run your application and you identify by your credentials. If you are logged in to your PC, no one else can log in at the same time.
The Terminal Server software takes away this restriction. But still Microsoft has licensed your OS to be used only by one person at the same time. 
So to stay within legal realms you have to buy an extra license for the number of users that should be able to log in in parallel.
The bot uses the credentials of a user on the computer. (He can log in with the same user multiple times in parallel or use multiple users in parallel)
Only one process can be run in one session. But one process can be run in multiple sessions.
Each session will use at least 2 GB RAM.

sebermuley replied:
I assume that if you buy a terminal server software they tell you to acquire user licenses for the number of users that should be able to log in in parallel. 
We do not maintain or (anymore) ship that software and should not advertise one.

sebermuley replied:
I agree that this is a bit vague, but I don't know how to precise that further within our limited universe.

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,12 +30,28 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+** We recommend using dedicated accounts and a user naming scheme to differentiate between local users and domain users and between human users and bot users.
+** User accounts used by the bot to run processes must belong to the _Remote Desktop Users_ group and must be allowed to establish a local RDP session via the local loopback address 127.0.0.2.
+* If you want a bot to run more than one (Windows 10, Windows 11) or two (Windows Server) Secure Sessions simultaneously, make sure that a terminal server software with enough licensed users is installed on the computer where the bot is installed. The computer must have enough resources to execute several processes simultaneously: 2 GB or more RAM per session. 
+* Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 
```
should "but" be "instead"?

hannanelson replied:
"licensed secure session" to match the next bullet

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -55,19 +61,20 @@ Format: `+http://<Serveraddress>:<Port>+`, for example: `+http://myproxy.de:881+
-Enable the bot to use secure sessions by entering credentials the bot uses to log in to the computer and to open a hidden session. If you don't add credentials here, the bot can only use a running desktop session.
+If you want to have a default user account, that the RPA Bot uses to log in to the session on the computer, where the bot runs processes, enter the credentials here. When users create a process run configuration in RPA Manager, they can choose this user or specify another there.
 
 image:bot-configurator-session-settings.png[The RPA Bot configurator application showing the Secure Session settings panel, 50%, 50%]
```
```suggestion
image:bot-configurator-session-settings.png[The RPA Bot Configurator application showing the Secure Session settings panel, 50%, 50%]
```

hannanelson replied:
(to match other mentions)

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -55,19 +61,20 @@ Format: `+http://<Serveraddress>:<Port>+`, for example: `+http://myproxy.de:881+
 
-Here you can define how RPA Bot components (Robot, StSec and StSem) communicate among themselves. These settings are relevant only for Secure Sessions. You can choose the Named Pipes protocol or TCP. Named Pipes is selected by default because it is a simple communication method. In some cases, when using TCP, the firewall can block the connection even though it is an exclusive internal computer p ...
-
 image:bot-configurator-secure-session.png[The RPA Bot configurator application showing the Secure Session Settings panel, 50%, 50%]
```
init cap?

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -78,13 +112,12 @@ After deinstallation or reconfiguration, a bot is still shown in the *Bots* view
-* xref::botmanagement-know.adoc[Getting to Know Bots]
-//* xref::botmanagement-manage.adoc[Managing Bots]
-* xref::botmanagement-troubleshoot.adoc[Troubleshooting Bots]
+* xref:rpa-home::automation-security-securesessions.adoc[Secure Sessions]
```
If the headings in the H1 are the same, remove the hard-coded links to autogenerate (and reduce maintenance).

sebermuley replied:
I agree, but would like to include the product name in the link text when I link to docs outside the current second navigation level.

hannanelson replied:
Is that the case here?

sebermuley replied:
no, but consistency gives me security :)
I will revisit taht with the renewal of the RPA content structure project...

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -16,29 +16,35 @@ After saving your changes, restart the RPA Bot Service for the changes to take e
 == Common Settings
 
-image:bot-configurator-common.png[The RPA Bot configurator application showing the Common settings panel, 50%, 50%]
+image:bot-configurator-common.png[The RPA Bot Configurator application showing the Common settings panel, 50%, 50%]
```
check other alt text for this change

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -16,29 +16,35 @@ After saving your changes, restart the RPA Bot Service for the changes to take e
+* *TEST CONNECTION*
++
+Click *TEST CONNECTION* to verify that the bot can establish a connection to RPA Manager.
 * *Alive interval*
```
suggest to the UI designers to make these items init cap (or at least consistent) :)

sebermuley replied:
@tasmoore Please check this for the next Bot (Configurator) release. There are more of these.... Thank you

sebermuley replied:
included in bug mentioned somewhere in this conversation

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -16,29 +16,35 @@ After saving your changes, restart the RPA Bot Service for the changes to take e
 +
 Time interval in milliseconds after which the bot connects to RPA Manager and checks for provided packages to collect (such as scheduling tasks or activities).
-* *Cleanup workflows on start*
+* *Cleanup activities on start*
```
Suggest to the UI designer to use "Clean up" (verb form) :)

sebermuley replied:
There is a bug for this already 
@tasmoore [(Bot) Bot Configurator - UI CX/UX Bugs](https://gus.lightning.force.com/a07EE00001cJLQOYA4)

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -16,29 +16,35 @@ After saving your changes, restart the RPA Bot Service for the changes to take e
+Select this option to delete all runtime packages that are stored locally in the bot. The process activities retrieve the required runtime packages from the RPA Manager on the next run. After the runtime packages are deleted, the Bot Configurator deselects this setting.
+* *Maximum number of sessions*
++
+Set the maximum number of Secure Sessions that the bot can use. Enabling this setting, limits but not overwrites the same setting in RPA Manager Bot Management.
```
```suggestion
Set the maximum number of secure sessions that the bot can use. Enabling this setting limits, but doesn't overwrite, the same setting in RPA Manager Bot Management.
```

hannanelson replied:
What does "limits... the same setting" mean? Does it limit the number of secure sessions but doesn't overwrite the setting elsewhere?

sebermuley replied:
Yes
Example
Bot Setting + Manager Setting = Valid Setting
42 + 1 = 1
1 + 42 = 1

hannanelson replied:
Can you revise to make clear?

Enabling this setting limits the number of secure sessions, but doesn't overwrite the same setting in RPA Manager Bot Management.

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -55,19 +61,20 @@ Format: `+http://<Serveraddress>:<Port>+`, for example: `+http://myproxy.de:881+
 == Session Settings
 
-Enable the bot to use secure sessions by entering credentials the bot uses to log in to the computer and to open a hidden session. If you don't add credentials here, the bot can only use a running desktop session.
+If you want to have a default user account, that the RPA Bot uses to log in to the session on the computer, where the bot runs processes, enter the credentials here. When users create a process run configuration in RPA Manager, they can choose this user or specify another there.
```
```suggestion
To specify a default user account that the RPA Bot uses to log in to the session and run processes on the computer, enter the credentials here. When users create a process-run configuration in RPA Manager, they can choose this user or specify another there.
```

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -55,19 +61,20 @@ Format: `+http://<Serveraddress>:<Port>+`, for example: `+http://myproxy.de:881+
-* *Default Secure Session User*
+* *Default Session User*
++
+If you want the bot to have a default user to log in to the session for the execution of a process, enter the credentials here. If the environment of the computer the bot is installed on is embedded in a domain and you want to use a local user, write the user name as follows: `.\username`. For domain users, use the following form: `mydomain\username`.
```
Can you use similar terminology as on line 64?

To specify that the RPA Bot use a default user account to log in to the session and run processes on the computer, enter the credentials here.

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -55,19 +61,20 @@ Format: `+http://<Serveraddress>:<Port>+`, for example: `+http://myproxy.de:881+
+* *Send images for process streaming*
 +
-If you want the bot to use Secure Sessions, enter the name of the user the bot uses to log in to the computer and to run a hidden session. If the environment of the computer the bot is installed on is embedded in a domain and you want to use a local user, write the user name as follows: `.\username`. For domain users, use the following form: `mydomain\username`.
+Enable this checkbox to transfer screenshots used for process streaming to RPA Manager. Disable the checkbox if not everyone on the process team is allowed to view the processed data. For the Process Streaming to work, the usage of the Windows Taskbar must be enabled in the process run configuration, too.
```
We use select and deselect for [checkboxes](https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#MuleSoftCXWritingStyleReference-%22select,%22%22choose%22).

hannanelson commented on rpa-bot/1.3/modules/ROOT/pages/configuration.adoc:
```diff
@@ -134,4 +141,6 @@ Sends analysis packages if the Activity Workflow terminates with OK.
 
-* xref:rpa-manager::botmanagement-overview.adoc#getting-started[Bot Management]
+* xref:rpa-manager::botmanagement-overview.adoc#getting-started[RPA Manager - Bot Management]
+* xref:rpa-manager::processautomation-deploy.adoc[RPA Manager - Deploying Automations]
```
Do you need the hard-coded link text?

sebermuley replied:
Added this to the Docs Redesign project

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+= Secure Sessions
+
+In RPA automations, a bot simulates a human user. The bot logs in to a system, sees the screen, and can interact with the UI via key strokes and mouse actions. That way, the bot can even interact with legacy systems, that do not provide an API.
```
```suggestion
In RPA automations, a bot simulates a human user. The bot logs in to a system, sees the screen, and can interact with the UI via keystrokes and mouse actions. The bot can even interact with legacy systems that do not provide an API.
```

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+In RPA automations, a bot simulates a human user. The bot logs in to a system, sees the screen, and can interact with the UI via key strokes and mouse actions. That way, the bot can even interact with legacy systems, that do not provide an API.
+
+The bot logs in to the computer, where it is installed. But, of course, it does not sit on a chair in front of the desktop. Nevertheless, if you would enter the room where the computer is, you could see what the bot is doing: The screen would show the open applications, the mouse would  move around and typing would appear in forms. In MuleSoft RPA, we call this a _Desktop Session_. To prevent una ...
```
```suggestion
The bot logs in to the computer where it is installed. But, of course, it does not sit on a chair in front of the desktop. Nevertheless, if you enter the room where the computer is, you can see what the bot is doing: The screen shows the open applications, the mouse moves around and typing appears in forms. In MuleSoft RPA, we call this a _Desktop Session_. To prevent unauthorized data access or manipulation, you must secure the physical room.
```

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+The bot logs in to the computer, where it is installed. But, of course, it does not sit on a chair in front of the desktop. Nevertheless, if you would enter the room where the computer is, you could see what the bot is doing: The screen would show the open applications, the mouse would  move around and typing would appear in forms. In MuleSoft RPA, we call this a _Desktop Session_. To prevent una ...
+
+Or you could use what we call _Secure Sessions_.
```
```suggestion
Or you can use what RPA calls _Secure Sessions_.
```

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+Or you could use what we call _Secure Sessions_.
+
+Secure Sessions are hidden Desktop Sessions. They run in the background and only the bot can access them. An additional advantage of the Secure Sessions is, that the bot can multiply itsself to run several process instances in parallel.
```
This sounds familiar. See https://github.com/mulesoft/docs-rpa/pull/293/files#r1376615143

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+Secure Sessions are hidden Desktop Sessions. They run in the background and only the bot can access them. An additional advantage of the Secure Sessions is, that the bot can multiply itsself to run several process instances in parallel.
+
+To use Secure Sessions you must ensure that all prerequisites are met and all responsible persons are informed.
```
You defined these as partials but still need to include them?
https://github.com/mulesoft/docs-rpa/pull/293/files#r1376618297

sebermuley replied:
will be part of the Doc Space Redesign project

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+== General Considerations
+
+* At the same time, a bot can only run in Desktop or in Secure Session mode.
+* You can not switch the session mode while configurations are deployed on the bot.
```
```suggestion
* You can't switch the session mode while configurations are deployed on the bot.
```

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+== General Considerations
+
+* At the same time, a bot can only run in Desktop or in Secure Session mode.
```
```suggestion
* A bot either can run in Desktop or Secure Session mode.
```

sebermuley replied:
Replaced with 
A bot either can run in Desktop Session or Secure Session mode. You can change the mode while no configurations are deployed to the bot.
[While - because it refers to a time span]

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+== General Considerations
+
+* At the same time, a bot can only run in Desktop or in Secure Session mode.
```
Is there a way to further group these bullets into logical sub-bullets or hanging paragraphs? Like session mode, others? They seem sort of random right now.

sebermuley replied:
This needs more thinking about... --> Postponed

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+* You can not switch the session mode while configurations are deployed on the bot.
+* Run configurations specify a session mode, too.
+* You can only deploy configurations to a bot that is in the same session mode.
+* You can specify the login data for the bot and the maximum number of Secure Sessions that the bot may run in parallel in the RPA Bot Configurator or in the RPA Manager.
```
```suggestion
* You can specify the login data for the bot and the maximum number of secure sessions that the bot can run in parallel in the RPA Bot Configurator or the RPA Manager.
```

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+* Run configurations specify a session mode, too.
+* You can only deploy configurations to a bot that is in the same session mode.
+* You can specify the login data for the bot and the maximum number of Secure Sessions that the bot may run in parallel in the RPA Bot Configurator or in the RPA Manager.
+** The maximum number of Secure Sessions set and enabled in the RPA Bot Configurator overwrites the same setting of RPA Manager.
```
```suggestion
+
The maximum number of secure sessions specified in the RPA Bot Configurator overwrites the same setting in RPA Manager.
```

hannanelson replied:
Previously, I had a question about the wording of something similar (limits).

hannanelson replied:
(Removed the single sub-bullet, per style)

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+* At the same time, a bot can only run in Desktop or in Secure Session mode.
+* You can not switch the session mode while configurations are deployed on the bot.
+* Run configurations specify a session mode, too.
```
Does the user specify the session mode for a run configuration? Or does the run configuration do that?

sebermuley replied:
The user specifies the mode for a run configuration and selects a fitting bot in one of the next steps.

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+* You can specify the login data for the bot and the maximum number of Secure Sessions that the bot may run in parallel in the RPA Bot Configurator or in the RPA Manager.
+** The maximum number of Secure Sessions set and enabled in the RPA Bot Configurator overwrites the same setting of RPA Manager.
+* Desktop Sessions:
+** The specified user must be logged in to this session to deploy a run configuration. 
```
should "this session" be "the current session" or "the same session"?

sebermuley replied:
The user must be logged in which  creates a session. This session is then overtaken by the bot.
This is already described somewhere...
--> Postponed

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+** The maximum number of Secure Sessions set and enabled in the RPA Bot Configurator overwrites the same setting of RPA Manager.
+* Desktop Sessions:
+** The specified user must be logged in to this session to deploy a run configuration. 
+** Do not close or minimize the console session while the bot executes a process.
```
Our style is to make bullet list items parallel. This bullet is an imperative instruction, but the previous one is just a requirement. Let me know if this is unclear.

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+** The specified user must be logged in to this session to deploy a run configuration. 
+** Do not close or minimize the console session while the bot executes a process.
+* Secure Sessions:
+** The higher the number of secure sessions the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.
```
```suggestion
** The higher the number of secure sessions, the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.
```

hannanelson replied:
Suggest an alternative way to word this - something like "The number of secure sessions can negatively affect bot performance. 

How much is "too much load"? 

Can you link to the requirements document here?

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+** Do not close or minimize the console session while the bot executes a process.
+* Secure Sessions:
+** The higher the number of secure sessions the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.
+** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 
```
See previous comment on this: https://github.com/mulesoft/docs-rpa/pull/293/files#r1376641014
It looks like you defined the partial, but didn't include it here?

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+** Login data for the bot
+** RPA Bot Configurator settings 
+* If you want to use Secure Sessions, read and pass on the following information:
+** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 
```
This is duplicated elsewhere. Can you include the partial?

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+* Secure Sessions:
+** The higher the number of secure sessions the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.
+** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 
+** Do not sign in as the specified user to a computer where a bot with one licensed secure session is running a process.
```
Duplicated in partials file - add include here?

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+* If you want to use Secure Sessions, read and pass on the following information:
+** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 
+** If you want to assign more than one secure session to a bot, make sure that a terminal server software is installed on the computer of the bot and refer to the xref:rpa-bot::hardware-software-requirements#requirements-for-secure-session-technology.adoc[system and hardware requirements documentation] to determine the maximum number of secure sessions for the bot.
+** Do not sign in to a computer where a bot with one licensed secure session is running a process.
```
Duplicated in partials file - add include here?

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+== Next Steps
+
+. In the documentation, check the System Requirements.
```
link

sebermuley replied:
linked in See Also
--> postponed to RPA Doc Space Reddisign

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+== Next Steps
+
+. In the documentation, check the System Requirements.
+. On the computer where you want to run automations and where the bot therefore is installed, check the system data:
```
```suggestion
. On the computer where you want to run automations, where the bot is installed, check the system data:
```

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
++
+* Operating system
+* RAM
+* Terminal Server software
```
Terminal server?

sebermuley replied:
it's a name

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+* Terminal Server software
+* Login data for the bot
+* RPA Bot Configurator settings
+. If you want to use default session settings for the run configuration, set the bot login data and the maximum number of parallel Secure Sessions in the Bot Configurator. 
```
Global - check to see if you can lowercase secure sessions, except when referring to the feature.

sebermuley replied:
uppercased all, because it'S always the feature

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+* RPA Bot Configurator settings
+. If you want to use default session settings for the run configuration, set the bot login data and the maximum number of parallel Secure Sessions in the Bot Configurator. 
+. In RPA Manager, enter the relevant data or a contact person in the description of the bot in the Bots view of the Bot Management module.
+.. If the bot should use Secure Sessions and the RPA Bot Configurator does not contain this setting, set the maximum number of parallel Secure Sessions that the bot may use via the Secure Session Assignment view.
```
```suggestion
+
If the bot should use Secure Sessions and the RPA Bot Configurator does not contain this setting, set the maximum number of parallel Secure Sessions that the bot can use via the Secure Session Assignment view.
```

hannanelson replied:
I noticed that you had a single sub-step. This should just be a hanging paragraph. Our style is to not have single un/ordered list items.

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+. If you want to use default session settings for the run configuration, set the bot login data and the maximum number of parallel Secure Sessions in the Bot Configurator. 
+. In RPA Manager, enter the relevant data or a contact person in the description of the bot in the Bots view of the Bot Management module.
+.. If the bot should use Secure Sessions and the RPA Bot Configurator does not contain this setting, set the maximum number of parallel Secure Sessions that the bot may use via the Secure Session Assignment view.
+. When creating a run configuration in the Process Automation module of RPA Manager, keep the general considerations in mind. 
```
link? 

Should this be something like "Review the General Considerations before..."?

hannanelson commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,50 @@
+
+== See Also
+
+* xref:rpa-bot::hardware-software-requirements.adoc[RPA Bot - Hardware and Software Requirements]
```
Global - avoid hard-coded link text if possible.

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+
+* Name and describe a bot.
+* Assign service times to a bot to determine their office hours.
+* Restrict bots to project phases
```
```suggestion
* Restrict bots to project phases.
```

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+* Name and describe a bot.
+* Assign service times to a bot to determine their office hours.
+* Restrict bots to project phases
+* Set the maximum number of sessions a bot may use.
```
```suggestion
* Set the maximum number of sessions a bot can use.
```

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+* Restrict bots to project phases
+* Set the maximum number of sessions a bot may use.
+
+If a bot is troubling you, you can do the following:
```
"troubling" is troubling me :) 

Can you be more specific? If a bot isn't working properly?

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
 
 * At least one bot must be installed on a separate computer.
-* If you want to assign more than one secure session to a bot, make sure that a terminal server software is installed on the computer of the bot.
+* On the computer where you want to run automations and where a bot therefore is installed, check the system data:
```
See previous similar edits. Opportunity for reuse?

sebermuley replied:
This needs more consideration (= too few time before release) --> Included in Doc Space Redesign project

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+** Terminal Server software
+** Login data for the bot
+** RPA Bot Configurator settings 
+* If you want to use Secure Sessions, read and pass on the following information:
```
What does "pass on" mean? Share with...whom?

sebermuley replied:
See above comment. We don't really know. Let's revisit that later.

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+** RPA Bot Configurator settings 
+* If you want to use Secure Sessions, read and pass on the following information:
+** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 
+** If you want to assign more than one secure session to a bot, make sure that a terminal server software is installed on the computer of the bot and refer to the xref:rpa-bot::hardware-software-requirements#requirements-for-secure-session-technology.adoc[system and hardware requirements documentation] to determine the maximum number of secure sessions for the bot.
```
Is this xref format correct?
`xref:rpa-bot::hardware-software-requirements#requirements-for-secure-session-technology.adoc`

hannanelson replied:
This content is similar to other content. Possible to consolidate/reuse?

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user. 
+** If you want to assign more than one secure session to a bot, make sure that a terminal server software is installed on the computer of the bot and refer to the xref:rpa-bot::hardware-software-requirements#requirements-for-secure-session-technology.adoc[system and hardware requirements documentation] to determine the maximum number of secure sessions for the bot.
+** Do not sign in to a computer where a bot with one licensed secure session is running a process.
+* If you want to use desktop sessions, read and pass on the following information:
```
Should desktop be init cap? 
Clarify "pass on"

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+** Do not sign in to a computer where a bot with one licensed secure session is running a process.
+* If you want to use desktop sessions, read and pass on the following information:
+** The specified user must be logged in to this session to deploy a run configuration. 
+** Do not close or minimize the console session while the bot executes a process.
```
These bullets aren't parallel.

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -24,15 +49,24 @@ Bots can be licensed to use secure sessions. Some properties, like name, service
-If bots should run processes in a hidden session, assign them secure session licenses.
+The bot runs a process configuration in a session on the computer where the bot is installed. 
+
+If bots should run processes in a hidden session, assign them at least one secure session.
```
```suggestion
To specify that a bot runs processes in a hidden session, assign the bot at least one secure session.
```

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -24,15 +49,24 @@ Bots can be licensed to use secure sessions. Some properties, like name, service
 The number of secure sessions that can be assigned and served by the bot depends on the hardware of the computer on which the bot is installed
 Please refer to the system and hardware requirements topic to assess the maximum number of secure sessions for the bot.
 
+Enabling the same setting via the RPA Bot Configurator, limits but not not overwrites this setting.
```
See comments here: https://github.com/mulesoft/docs-rpa/pull/293/files#r1376649413

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -24,15 +49,24 @@ Bots can be licensed to use secure sessions. Some properties, like name, service
+
 . In the *Bots* view of the *Bot Management* module, click *Secure Session Assignment*.
-. Set the number of sessions the bot may use.
+. Set the number of sessions the bot may use:
```
```suggestion
. Set the number of sessions the bot can use:
```

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -24,15 +49,24 @@ Bots can be licensed to use secure sessions. Some properties, like name, service
+. Set the number of sessions the bot may use:
++
+* Session Assignment >= 1: 
+** The bot consumes up the the specified number of Secure Sessions.
```
```suggestion
** The bot consumes up to the specified number of secure sessions.
```

hannanelson replied:
Check the content with vale - I think it flags common double words.

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -24,15 +49,24 @@ Bots can be licensed to use secure sessions. Some properties, like name, service
+* Session Assignment >= 1: 
+** The bot consumes up the the specified number of Secure Sessions.
+* Session Assignment = 0
+The bot consumes the desktop session.
```
Is this correct? It sounds like it's eating the session! Should desktop be init cap? 
Should it be `** The bot...`

hannanelson commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -78,13 +112,12 @@ After deinstallation or reconfiguration, a bot is still shown in the *Bots* view
 . Confirm the removal.
 
-If you accidentally remove a bot that is still being executed, the bot will shut down and the API Key is deleted on the bot's computer. You must uninstall and reinstall the bot with a new API Key. In this case, the licenses, name, description, assigned process applications and service times, and tasks will be lost and have to be assigned again.
+If you accidentally remove a bot that is still being executed, the bot shuts down and the API Key is deleted on the bot's computer. You must uninstall and reinstall the bot with a new API Key. In this case, the licenses, name, description, assigned process applications and service times, and tasks are lost and have to be assigned again.
```
```suggestion
If you accidentally remove a bot that is executing, the bot shuts down and the API Key is deleted on the bot's computer. You must uninstall and reinstall the bot with a new API Key. In this case, you must re-assign the licenses, name, description, assigned process applications and service times, and tasks.
```

hannanelson commented on rpa-manager/modules/ROOT/pages/processmonitoring-stream.adoc:
```diff
@@ -18,6 +18,7 @@ In the *Process Streaming* view, you see the number of configurations, RPA Bots,
 
 * Only productive processes for which at least one configuration has been created and for which you are a member of the process team are displayed.
 * Check xref:processautomation-deploy.adoc#runconfig-execution-enabletaskbar[*Enable Taskbar (Start menu, taskbar buttons)*] in the run configuration to view the desktop of the computer that runs the process. Otherwise, a black screen is shown.
+* xref:rpa-bot::configuration.adoc#session-settings[*Send images for process streaming*] must be enabled in the RPA Bot Configurator on the computer, where the process runs.
```
```suggestion
* You must enable xref:rpa-bot::configuration.adoc#session-settings[*Send images for process streaming*] in the RPA Bot Configurator on the computer where the process runs.
```

sebermuley replied:
The reader cannot enable this himself

Commit:
implemented suggestions from Editor review - 1
(Bot System Requirements)

Commit:
Suggestions from editorial review - bot config

Commit:
...

Commit:
...

Commit:
Merge branch 'latest' into bypassmergelatest

Commit:
implemented last suggestions from editorial review

tasmoore commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,29 +30,45 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+
+=== Requirements for the Secure Session Technology
+
+Secure Sessions are hidden Desktop sessions that run in the background and are accessible only to the bot. An additional advantage of using Secure Sessions is, that the bot can multiply itself to run several process instances in parallel.
```
```suggestion
Secure Sessions are hidden Desktop sessions that run in the background and are accessible only to the bot. An additional advantage of using Secure Sessions is that the bot can multiply itself to run several process instances in parallel.
```

tasmoore commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,53 @@
+* You can specify the login data for the bot and the maximum number of Secure Sessions that the bot can run in parallel in the RPA Bot Configurator or in the RPA Manager.
+** The maximum number of Secure Sessions specified in the RPA Bot Configurator limits overwrites the same setting in RPA Manager.
++
+Enabling the same setting via the RPA Bot Configurator, limits but not not overwrites this setting.
```
```suggestion
Enabling the same setting via the RPA Bot Configurator limits but does not overwrite this setting.
```

sebermuley replied:
Replaced with Hanna's suggestion from another file (organizing these informations and implementimng them as partials will be part of the Doc Space Redesign project.)

tasmoore commented on rpa-home/modules/ROOT/pages/automation-security-securesessions.adoc:
```diff
@@ -0,0 +1,53 @@
+** The console session must remain open and unminimized while the bot executes a process.
+* Secure Sessions:
+** The higher the number of Secure Sessions the lower the performance of the bot. Too much load can cause process slowdown or disruptions. Check the system data and the system requirements documentation to determine the maximum number of sessions that the bot can safely execute in parallel.
+** Before deploying a run configuration on a bot that has only one Secure Session licensed, do not disconnect the session, but sign out the specified user. 
```
```suggestion
** Before deploying a run configuration on a bot that has only one Secure Session licensed, do not disconnect the session but do sign out the specified user. 
```

sebermuley replied:
Replaced all occurrences with suggestions from Hanna

tasmoore commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -1,11 +1,36 @@
+** Login data for the bot
+** RPA Bot Configurator settings
+* If you want to use Secure Sessions, read and pass on the following information:
+** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session, but sign out the specified user.
```
```suggestion
** Before deploying a run configuration on a bot that has only one secure session licensed, do not disconnect the session but do sign out the specified user.
```

sebermuley replied:
see above

tasmoore commented on rpa-manager/modules/ROOT/pages/botmanagement-manage.adoc:
```diff
@@ -78,13 +112,12 @@ After deinstallation or reconfiguration, a bot is still shown in the *Bots* view
+* xref:rpa-home::automation-security-securesessions.adoc[RPA - Secure Sessions]
+* xref:rpa-bot::hardware-software-requirements.adoc[RPA Bot - Hardware and Software Requirements]
+* xref:rpa-bot::configuration.adoc[RPA Bot Configurator]
+* xref:botmanagement-troubleshoot.adoc[Trouble-shooting Bots]
```
```suggestion
* xref:botmanagement-troubleshoot.adoc[Troubleshooting Bots]
```

tasmoore commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,29 +30,45 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+** Use dedicated accounts and a user-naming scheme to differentiate between local users and domain users and between human users and bot users.
+** User accounts used by the bot to run processes must belong to the `Remote Desktop Users` group and must have permission to establish a local RDP session via the local loopback address 127.0.0.2.
+* To enable a bot to run more than one (Windows 10, Windows 11) or two (Windows Server) Secure Sessions simultaneously, ensure that a terminal server software with enough licensed users is installed on the computer where the bot is installed. The computer must have enough resources to execute several processes simultaneously: 2 GB or more RAM per session.
+* Before deploying a run configuration on a bot that has only one licensed secure session, do not disconnect the session, instead sign out the specified user.
```
```suggestion
* Before deploying a run configuration on a bot that has only one licensed Secure Session, do not disconnect the session, instead sign out the specified user.
```

tasmoore commented on rpa-bot/1.3/modules/ROOT/pages/_partials/hard-soft-requirements.adoc:
```diff
@@ -28,29 +30,45 @@ include::rpa-home::partial$windows-versions-support.adoc[]
+** User accounts used by the bot to run processes must belong to the `Remote Desktop Users` group and must have permission to establish a local RDP session via the local loopback address 127.0.0.2.
+* To enable a bot to run more than one (Windows 10, Windows 11) or two (Windows Server) Secure Sessions simultaneously, ensure that a terminal server software with enough licensed users is installed on the computer where the bot is installed. The computer must have enough resources to execute several processes simultaneously: 2 GB or more RAM per session.
+* Before deploying a run configuration on a bot that has only one licensed secure session, do not disconnect the session, instead sign out the specified user.
+** Do not sign in to a computer where a bot with one licensed secure session is running a process.
```
```suggestion
** Do not sign in to a computer where a bot with one licensed Secure Session is running a process.
```

tasmoore reviewed (changes requested):
Hi Simone, I added a few suggestions. Please take a look.

Commit:
implemented suggestions from peer review

sebermuley committed:
Merge branch 'latest' into bypassmergelatest

sebermuley commented:
Hi @tasmoore Thank you. I implemented your suggestions (or replaced them with Hanna's) and updated the branch. We can revisit the topics after the next Bot release.
